src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReportingService.java

```diff
@@ -0,0 +1,173 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import org.postgresql.Driver;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskHistoryResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.replicarepository.TaskHistoryResourceRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.repository.TaskResourceRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.exceptions.ReplicationException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Properties;
+import javax.sql.DataSource;
+
+
+@Service
+public class MIReportingService {
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(MIReportingService.class);
+    public static final String MAIN_SLOT_NAME = "main_slot_v1";
+
+    String user;
+    String password;
+
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    @Qualifier("replicaDataSource")
+    private DataSource replicaDataSource;
+
+    private final TaskHistoryResourceRepository taskHistoryRepository;
+    private final TaskResourceRepository taskResourceRepository;
+
+
+    public MIReportingService(TaskHistoryResourceRepository tasksHistoryRepository,
+                              TaskResourceRepository taskResourceRepository,
+                              @Value("${replication.username}") String user,
+                              @Value("${replication.password}") String password) {
+        this.taskHistoryRepository = tasksHistoryRepository;
+        this.taskResourceRepository = taskResourceRepository;
+        this.user = user;
+        this.password = password;
+    }
+
+    public List<TaskHistoryResource> findByTaskId(String taskId) {
+        return taskHistoryRepository.getByTaskId(taskId);
+    }
+
+    public void logicalReplicationCheck() {
+        LOGGER.debug("Postgresql logical replication check executed");
+        if (isReplicationSlotPresent()) {
+            if (!isPublicationPresent()) {
```

Comment:
As the comment above.  Could you get multiple threads attempting to do this at the same time?  Will this have unwanted side effects?